perun-core/src/main/java/cz/metacentrum/perun/core/impl/UsersManagerImpl.java

```diff
@@ -1378,4 +1387,65 @@ public void deleteSponsorLinks(PerunSession sess, User sponsor) throws InternalE
 		}
 
 	}
+
+	@Override
+	public List<Resource> getAssignedResources(PerunSession sess, User user) {
+		try  {
+			return jdbc.query("select distinct " + resourceMappingSelectQuery + " from resources join groups_resources on resources.id=groups_resources.resource_id " +
+					" join groups on groups_resources.group_id=groups.id" +
+					" join groups_members on groups.id=groups_members.group_id " +
+					" join members on groups_members.member_id=members.id " +
+					" where members.user_id=?", RESOURCE_MAPPER, user.getId());
+		} catch (RuntimeException e) {
+			throw new InternalErrorException(e);
+		}
+	}
+
+	@Override
+	public List<Resource> getAllowedResources(PerunSession sess, User user) {
+		try  {
+			return jdbc.query("select distinct " + resourceMappingSelectQuery + " from resources join groups_resources on resources.id=groups_resources.resource_id " +
+					" join groups on groups_resources.group_id=groups.id" +
+					" join groups_members on groups.id=groups_members.group_id " +
+					" join members on groups_members.member_id=members.id " +
+					" where members.user_id=? and members.status!=? and members.status!=?",
+					RESOURCE_MAPPER, user.getId(), String.valueOf(Status.INVALID.getCode()), String.valueOf(Status.DISABLED.getCode()));
+		} catch (RuntimeException e) {
+			throw new InternalErrorException(e);
+		}
+	}
+
+	@Override
+	public List<Resource> getAssignedResources(PerunSession sess, Facility facility, User user) {
+		try {
+			return jdbc.query("select distinct " + ResourcesManagerImpl.resourceMappingSelectQuery + " from resources "+
+							" join groups_resources on groups_resources.resource_id=resources.id" +
+							" join groups_members on groups_members.group_id=groups_resources.group_id" +
+							" join members on members.id=groups_members.member_id" +
+							" where resources.facility_id=? and members.user_id=?",
+					RESOURCE_MAPPER, facility.getId(), user.getId());
+		} catch (RuntimeException e) {
+			throw new InternalErrorException(e);
+		}
+	}
+
+	@Override
+	public List<RichResource> getAssignedRichResources(PerunSession sess, User user) {
+		try  {
+			return jdbc.query("select distinct " + resourceMappingSelectQuery + ", " + VosManagerImpl.voMappingSelectQuery + ", " +
+					FacilitiesManagerImpl.facilityMappingSelectQuery + ", "+resourceTagMappingSelectQuery+" from resources" +
+					" join vos on resources.vo_id=vos.id " +
```

Comment:
I prefer it before leaving space at the each line end, as we previously did. Its more visible and prevents modifying SQL the way it would fail (having no space between keywords).